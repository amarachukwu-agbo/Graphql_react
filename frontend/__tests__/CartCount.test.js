import { shallow } from 'enzyme';
import CartCount from '../components/CartCount';

describe('<CartCount />', () => {
  it('renders properly', () => {
    const wrapper = shallow(<CartCount count={2}/>);
    expect(wrapper.length).toBe(1);
  });
  it('matches snapshot', () => {
    const wrapper = shallow(<CartCount count={2}/>);
    expect(wrapper).toMatchSnapshot();
  });
  it('updates with the props', () => {
    const wrapper = shallow(<CartCount count={50}/>);
    wrapper.setProps({
      count: 2
    })
    expect(wrapper).toMatchSnapshot();
  });
});