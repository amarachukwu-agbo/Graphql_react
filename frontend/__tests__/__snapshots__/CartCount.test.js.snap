// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> matches snapshot 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="count"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Count>
        2
      </CartCount__Count>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCount /> updates with the props 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="count"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Count>
        2
      </CartCount__Count>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;
